packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,298 @@ export class BitmapFont
+     *
+     * const title = new PIXI.BitmapText("This is the title", { font: "TitleFont" });
+     */
+    public static from(options: IBitmapFontFactoryOptions): BitmapFont
+    {
+        const name = options.name;
+        let chars = options.chars;
+
+        if (name && BitmapFont.available[name])
+        {
+            throw new Error(`[Name Collision]: The name "${name}" has already been taken by another bitmap font`);
+        }
+
+        // Split the chars string into individual characters
+        if (typeof chars === 'string')
+        {
+            chars = chars.split('');
+        }
+        // Handle an array of characters+ranges
+        else if (chars.find((elem) => Array.isArray(elem)))
+        {
+            const flatChars = [];
+
+            for (let i = 0, j = chars.length; i < j; i++)
+            {
+                const elem = chars[i];
+
+                // Handle range delimited by start/end chars
+                if (Array.isArray(elem))
+                {
+                    if (elem.length !== 2)
+                    {
+                        let suffix;
+
+                        switch (i)
+                        {
+                            case 1: suffix = 'st'; break;
+                            case 2: suffix = 'nd'; break;
+                            case 3: suffix = 'rd'; break;
+                            default: suffix = 'th';
+                        }
+
+                        throw new Error('BitmapFont.from was provided a character range is more than two limits. '
+                            + `This is the ${i}${suffix} input in the character set.`);
+                    }
+
+                    for (let i = elem[0].charCodeAt(0), j = elem[1].charCodeAt(0); i <= j; i++)
+                    {
+                        flatChars.push(String.fromCharCode(i));
+                    }
+                }
+                // Handle a character set string
+                else
+                {
+                    flatChars.push(...elem.split(''));
+                }
+            }
+
+            chars = flatChars;
+        }
+
+        const resolution = options.resolution || 1;
+        const style = new TextStyle(options);
+
+        const lineWidth = BMT_DIMEN;
+
+        const fontData = new BitmapFontData();
+
+        fontData.info[0] = {
+            face: style.fontFamily,
+            size: style.fontSize,
+        };
+        fontData.common[0] = {
+            lineHeight: style.fontSize,
+        };
+        fontData.page[0] = {
+            id: 0,
+            file: '',
+        };
+
+        const textures: Texture[] = [];
+
+        let positionX = 0;
+        let positionY = 0;
+
+        let canvas;
+        let context;
+        let baseTexture;
+        let maxCharHeight = 0;
+
+        for (let i = 0; i < chars.length; i++)
+        {
+            if (positionY >= BMT_DIMEN - ((style.fontSize + BMT_PADDING) * resolution))
```

Comment:
It'd be nice to know, as a dev, how efficient we've been with the atlases.  If all we've added is numbers, then we could have a massive area not used, and we could reduce the size of the atlas. Or if we've used 3 or 4 atlases, we could know to increase the atlas size to say 2048x2048 and beyond. Not sure if it's beyond the scope of this PR or not...  but I know as soon as I start using the extra functionality here, it's gonna be one of the first questions I'm going to have!

Reply:
Is there any reason we couldn’t trim the canvas after we render glyphs to it?

Reply:
There is the canvas trim util we have; so could default to 2048x2048, to fit as much in as possible at a size we know all devices support, then trim down anything unused at the end?

Reply:
@themoonrat Where is the canvas util you're referring to?

Reply:
PIXI.utils.trimCanvas